src/interchange/src/json.rs

```diff
@@ -377,16 +381,25 @@ fn build_row_schema_fields(
     columns: &[(ColumnName, ColumnType)],
     type_namer: &mut Namer,
     custom_names: &BTreeMap<GlobalId, String>,
+    set_null_defaults: bool,
 ) -> Vec<serde_json::Value> {
     let mut fields = Vec::new();
     let mut field_namer = Namer::default();
     for (name, typ) in columns.iter() {
         let (name, _seen) = field_namer.valid_name(name.as_str());
-        let field_type = build_row_schema_field(type_namer, custom_names, typ);
-        fields.push(json!({
-            "name": name,
-            "type": field_type,
-        }));
+        let field_type = build_row_schema_field(type_namer, custom_names, typ, set_null_defaults);
+        if set_null_defaults && typ.nullable {
```

Comment:
In the future, we might change how we decide whether to make Avro schema fields nullable (i.e., we may not base it on `typ.nullable`). In particular, because of the work on nullability control that we plan to do soon after this.

So I'd suggest not checking `typ.nullable` but instead inspecting `field_type` directly to see whether it is an array whose first element is the string `"null"` (this is what nullable types look like in Avro schemas).

Reply:
👍

Reply:
That's its own type of brittle though, no? Like, what if something changes in how we emit field types and we go `["string", "null"]` instead of `["null", "string"]`? Thinking about a future world where we allow users to set per-field defaults, and in those cases you need the type of the default value to come first in the union.

Maybe a good approach here would be to `soft_assert!` that if field type is an array, it is an array with exactly two elements and the first element is `"null"`. I think that's the only type of Avro union type we should ever be emitting at the moment, right? And that way, if someone changes `build_row_schema_field` to do something different, things will loudly fail in CI.

Reply:
True. And yes, that's the only type of avro union we emit at this point (even if a source has a union type, it's already split into two separate columns). 

Going with `soft_assert!` here. When we implement the support for user defaults I think we'll have to refactor here a bit. The `build_row_schema_field` is technically returning the type not the entire field, which is why the logic is split now.

Reply:
Actually thinking more here, if we are checking that the first element is "null", it would be safe to set the default as "null". Even if say the types are reversed later, like `["string", "null"]`, we will not be setting `null` as default. (And this would only occur when user has provided a default value, so that will be set instead).

Reply:
Updated to checking if it's union type, also asserting that the first field is "null".

Reply:
> That's its own type of brittle though, no

We can't. The default must apply to the first field of the union, according to the spec. If someone makes that change then it's simply a bug.